homeassistant/components/anova/__init__.py

```diff
@@ -0,0 +1,91 @@
+"""The Anova integration."""
+from __future__ import annotations
+
+import logging
+
+from anova_wifi import (
+    AnovaApi,
+    AnovaPrecisionCooker,
+    AnovaPrecisionCookerSensor,
+    InvalidLogin,
+    NoDevicesFound,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN
+from .coordinator import AnovaCoordinator
+from .models import AnovaData
+from .util import serialize_device_list
+
+PLATFORMS = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Anova from a config entry."""
+    api = AnovaApi(
+        aiohttp_client.async_get_clientsession(hass),
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+    )
+    try:
+        await api.authenticate()
+        online_devices = await api.get_devices()
```

Comment:
Maybe pass in the cached devices so you don't have to create the objects here as well

Reply:
Might be better to pass them to the constructor